core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -263,7 +264,8 @@ private[spark] class IndexShuffleBlockResolver(
             file.delete()
           }
           if (!fileTmp.renameTo(file)) {
-            throw new IOException(s"fail to rename file ${fileTmp} to ${file}")
+            throw SparkException.internalError(
```

Comment:
@bozhang2820 Are you sure that IO errors should be considered as internal errors? Users can face to them in real life, I guess.

Reply:
Sorry for the late reply. Yes I think we might need a `SparkIOException` which extends `IOException`?

Reply:
I think that re-using the existing one `SparkRuntimeException` with a proper error class should be enough but `SparkIOException` works to me as well.